raft/src/test/java/org/apache/kafka/raft/MockLogTest.java

```diff
@@ -370,6 +365,23 @@ public void testReadOutOfRangeOffset() {
             Isolation.UNCOMMITTED));
     }
 
+    @Test
+    public void testMonotonicEpochStartOffset() {
+        appendBatch(5, 1);
+        assertEquals(5L, log.endOffset().offset);
+
+        log.initializeLeaderEpoch(2);
+        assertEquals(Optional.of(new OffsetAndEpoch(5L, 1)), log.endOffsetForEpoch(1));
+        assertEquals(Optional.of(new OffsetAndEpoch(5L, 2)), log.endOffsetForEpoch(2));
+
+        // Initialize a new epoch at the same end offset. The epoch cache ensures
+        // that the start offset of each retained epoch increases monotonically.
```

Comment:
I thought this is a test for end offset?

Reply:
The epoch cache tracks tuples of (epoch, start offset). The start offset of a new leader epoch is the current log end offset.